File: BE_QuanLyTro/src/main/java/ut/edu/be_quanlytro/Dto/Response/PaymentQrResponse.java
package ut.edu.be_quanlytro.Dto.Response;

import lombok.Builder;
import lombok.Data;
import java.math.BigDecimal;

@Data
@Builder
public class PaymentQrResponse {
    private String bankId;         // Tên viết tắt ngân hàng (VD: MB, VCB)
    private String accountNo;      // Số tài khoản
    private String accountName;    // Tên chủ tài khoản
    private BigDecimal amount;     // Số tiền cần chuyển
    private String content;        // Nội dung chuyển khoản
    private String qrImageUrl;     // Link cái ảnh QR để Frontend đem in ra
}

File: Flutter_QuanLyTro/lib/data/models/response/payment_qr_response.dart
class PaymentQrResponse {
  final String bankId;
  final String accountNo;
  final String accountName;
  final double amount;
  final String content;
  final String qrImageUrl;

  PaymentQrResponse({
    required this.bankId,
    required this.accountNo,
    required this.accountName,
    required this.amount,
    required this.content,
    required this.qrImageUrl,
  });

  factory PaymentQrResponse.fromJson(Map<String, dynamic> json) {
    return PaymentQrResponse(
      bankId: json['bankId'] ?? '',
      accountNo: json['accountNo'] ?? '',
      accountName: json['accountName'] ?? '',
      amount: (json['amount'] ?? 0).toDouble(),
      content: json['content'] ?? '',
      qrImageUrl: json['qrImageUrl'] ?? '',
    );
  }
}